src/server/scripts/EasternKingdoms/ShadowfangKeep/shadowfang_keep.cpp

```diff
@@ -93,7 +91,7 @@ class npc_shadowfang_prisoner : public CreatureScript
         InstanceScript* instance = creature->GetInstanceScript();
 
         if (instance && instance->GetData(TYPE_FREE_NPC) != DONE && instance->GetData(TYPE_RETHILGORE) == DONE)
-            player->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, GOSSIP_ITEM_DOOR, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 1);
+            player->ADD_GOSSIP_ITEM_DB(Player::GetDefaultGossipMenuForSource(creature), 0, GOSSIP_SENDER_MAIN, GOSSIP_OPTION_GOSSIP);
```

Comment:
Yes, this is how it should be. Look at the definition:

#define ADD_GOSSIP_ITEM_DB(h, i, c, d)   PlayerTalkClass->GetGossipMenu().AddMenuItem(h, i, c, d)

// h - Menu ID from DB
// i - Menu item ID from DB
// c - Sender(this is to identify the current Menu with this item)
// d - Action (identifys this Menu Item)